src/mpid/ch4/generic/mpidig_recv.h

```diff
@@ -20,13 +20,17 @@
 #define FUNCNAME MPIDIG_prepare_recv_req
 #undef FCNAME
 #define FCNAME MPL_QUOTE(FUNCNAME)
-static inline int MPIDIG_prepare_recv_req(void *buf, MPI_Aint count, MPI_Datatype datatype,
+static inline int MPIDIG_prepare_recv_req(int rank, int tag, MPIR_Context_id_t context_id,
```

Comment:
Actually I'm trying to remove `MPIDIG_prepare_recv_req` in #3590. Do you think this is going to be useful in other places?

Reply:
Yes, I am using it in the XPMEM PR #3452 . I need to create AM receive request in SHM:
https://github.com/pmodels/mpich/blob/8e4e8408244b0e407abd1de38c1ebccb04ca0441/src/mpid/ch4/shm/src/shm_p2p.h#L343-L355

Reply:
@hajimefu Do you think we can go ahead and merge this PR ?

Reply:
Yes. I'll drop my patch to remove `MPIDIG_prepare_recv_req`.